UnityProject/Assets/Scripts/Player/TeleportUtils.cs

```diff
@@ -111,5 +111,65 @@ public static void TeleportLocalGhostTo(Vector3 vector)
 		{
 			PlayerManager.LocalPlayerScript.playerNetworkActions.CmdGhostPerformTeleport(vector);
 		}
+
+		/// <summary>
+		/// Server: instantly teleports the given object to a random point on a circle centred on the object's current position.
+		/// </summary>
+		/// <param name="objectToTeleport">The object that is being teleported.</param>
+		/// <param name="minRadius">The minimum distance the object could teleprot to.</param>
+		/// <param name="maxRadius">The maximum distance the object could teleport to.</param>
+		/// <param name="tryAvoidSpace">Will perform (limited) rerolling of the random vector generation
+		/// until a tile that is not space is found. If the limit is reached, will default to the last space roll.</param>
+		/// <returns>The teleported object's new position.</returns>
+		public static Vector3Int ServerTeleportRandom(
+				GameObject objectToTeleport, int minRadius = 0, int maxRadius = 16, bool tryAvoidSpace = false)
+		{
+			Vector3Int originalPosition = objectToTeleport.RegisterTile().WorldPositionServer;
+			Vector3Int newPosition;
+
+			if (tryAvoidSpace)
+			{
+				newPosition = GetTeleportPosAvoidSpace(originalPosition, minRadius, maxRadius);
```

Comment:
Might be an issue, if it does fail to get a Pos which avoids space, it will go to 0,0. Might be better to just make it go to original pos instead

Reply:
Oh wait misread the code :p